### PR TITLE
Made nghttp2 link openssl and zlib using pinnings

### PR DESCRIPTION
**Pull request by mariusvniekerk**
Checklist
* [ ] Used a [fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -10,7 +10,7 @@ source:
   sha256: eacc6f0f8543583ecd659faf0a3f906ed03826f1d4157b536b4b385fe47c5bb8
 
 build:
-  number: 0
+  number: 1
   skip: true  # [not unix]
```
Comment (wolfv) [inline recipe/meta.yaml:15]: we should have a run_exports?

Comment (wolfv) [reply recipe/meta.yaml:15]: ah we have it in the outputs.

File: recipe/meta.yaml [hunk 2]
```diff
@@ -38,13 +38,13 @@ outputs:
...
         - libev >=4.11
         - c-ares >=1.7.5
       run:
-        - zlib >=1.2.3
-        - openssl >=1.0.1  # [unix]
+        - zlib
+        - openssl  # [unix]
         - libev >=4.11
         - c-ares >=1.7.5
         - libstdcxx-ng  # [linux]
```
Comment (wolfv) [inline recipe/meta.yaml:50]: isn't this added from the compiler stuff on linux anyways?

File: recipe/meta.yaml (hunk 1 above)
Comment (mariusvniekerk) [reply recipe/meta.yaml:15]: there is one already on line 53

**Comment by mariusvniekerk** [comment]
This should unblock https://github.com/conda-forge/curl-feedstock/pull/60

**Comment by wolfv** [comment]
looks like this could need a rerender?

**Comment by mariusvniekerk** [comment]
it had a rerender, not sure why the ci isn't triggering

File: recipe/meta.yaml (hunk 2 above)
Comment (mariusvniekerk) [reply recipe/meta.yaml:50]: turns out its not?

**Comment by wolfv** [comment]
@cpaulik do you want to merge this? Then we can move on with cURL

**Comment by cpaulik** [comment]
Thanks. I'm happy to merge this but I really don't know much about nghttp2. Was kind of pulled into this because of wanting it for Curl and gdal.

**Pull request merged**